sktime/covariance/covariance.py

```diff
@@ -135,6 +135,25 @@ def symmetrized(self) -> bool:
         """
         return self._symmetrized
 
+    def whiten(self, data: np.ndarray) -> np.ndarray:
+        r"""Whiten a (T, N)-shaped chunk of data by transforming it into the PCA basis. In case of rank deficiency
```

Comment:
in the case of a rank deficiency it doesn't always reduce the dimension?

Reply:
it does because we perform a truncated inverse of the covariances if they are not full rank

Reply:
so then why do the docs say it "might"?

Reply:
true